### PR TITLE
Opacus and Plan use case demo

### PR DESCRIPTION
**Pull request by xutongye**
## Description
This PR adds a notebook to demonstrate a use case with Opacus and Plan. The notebook is heavily based on #5230 .

And also, this PR fix the `VerifyAll/VerifyAll()` inconsistancy, by introducing a constance `VERIFYALL`.

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- examples/experimental/tongye/opacus_with_Plan/

## Checklist
- [ ] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [ ] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb [hunk 1]
```diff
@@ -0,0 +1,627 @@
+{
```
Comment (Darktex) [inline examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>nit: spelling ("trainning")</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/5333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='8'/>

Comment (Darktex) [inline examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>spelling: <code>lenth</code>  -&gt; <code>length</code>  <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/5333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

Comment (Darktex) [inline examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>But this is the pointer to the <strong>Dataset</strong>, not the dataloader. This length is the total number of examples, whereas if we check the length of the dataloader, we'd get the total number of batches.</p><p> </p><p>If we want to pass a pointer to the dataset instead that is totally fine, but then we should change the text cells to reflect that it's not in fact a dataloader :) </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/5333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Comment (Darktex) [inline examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>The super long duration might be an artifact of passing the dataset length instead of the dataloader length: in this case, batch size is 64 so there's going to be 64x the batches :D </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/5333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='12'/>

Comment (Darktex) [inline examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>Why not use real data since we have the pointer anyway? We can still limit the batches to the first 100</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/5333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='13'/>

Comment (Darktex) [inline examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>line 38: this is more pythonic: <code>data, target = data_tuple</code>.  You could also do this in the for loop directly:</p><pre class="ql-syntax" spellcheck="false">for _batch_idx, (data, target) in enumerate(dl):
</pre><p> </p><p>Same for line 48:</p><p> </p><p><code>eps, best_alpha = privacy_engine_ptr.get_privacy_spent(args["delta"])</code></p><p> <span class="ql-cursor">﻿</span></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/OpenMined/PySyft/pull/5333/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (xutongye) [reply examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>yeah, thanks</p>

Comment (xutongye) [reply examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>thanks</p>

Comment (xutongye) [reply examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>right, it's a typo, I did want length of dataloader, it should be <code>dl_length=train_loader_ptr.<strong>len</strong>()</code></p>

Comment (xutongye) [reply examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>yeah, we will see after fix the dataset length typo</p>

Comment (xutongye) [reply examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>It's how you use the Plan api for the moment. When define a function as a Plan, you should give default values to it's arguments, and the defaut value should be local, serializable, and sendable. So, we don't have the data locally, and a DataLoader is not serializable and sendable for the moment (we are trying to support that in #5314). So I use a List to fake it.</p>

Comment (xutongye) [reply examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>thanks</p>

Comment (madhavajay) [reply examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>Yes this confusing part of the API is high on the change list. 😊</p>

Comment (madhavajay) [reply examples/experimental/tongye/opacus_with_Plan/Opacus_Syft_Data_Scientist.ipynb:1]: <p>Sorry, this is a hangover from before we had Tuples and destructuring so its not intentional, we need to make sure this still works as expected because destructuring might be a little bit magic..... Actually I just tested and while destructuring does work it also seems to cause some small issue with the GC, so ill go make a ticket for that now. Thanks. 😊</p>

**Comment by madhavajay** [comment]
We are going to re-do this with the new serializable torch.nn.Module and the improved Plans once its available.
